Makefile

```diff
@@ -22,26 +28,20 @@ all: common otelcontribcol
 e2e-test: otelcontribcol
 	$(MAKE) -C testbed runtests
 
-.PHONY: precommit
-precommit:
-	$(MAKE) gotidy
-	$(MAKE) ci
-
 .PHONY: test-with-cover
-test-with-cover:
+unit-tests-with-cover:
 	@echo Verifying that all packages have test files to count in coverage
 	@scripts/check-test-files.sh $(subst github.com/open-telemetry/opentelemetry-collector-contrib/,./,$(ALL_PKGS))
-	@echo pre-compiling tests
-	set -e; for dir in $(ALL_MODULES); do \
-	  echo "go test ./... + coverage in $${dir}"; \
-	  (cd "$${dir}" && \
-	    $(GOTEST) $(GOTEST_OPT_WITH_COVERAGE) ./... && \
-	 	go tool cover -html=coverage.txt -o coverage.html ); \
-	done
+	@$(MAKE) for-all CMD="make do-unit-tests-with-cover"
+
+.PHONY: integration-tests-with-cover
+integration-tests-with-cover:
+	@echo $(INTEGRATION_TEST_MODULES)
+	@$(MAKE) for-all CMD="make do-integration-tests-with-cover" ALL_MODULES="$(INTEGRATION_TEST_MODULES)"
```

Comment:
I see we've added build tag to the integration test below. Do we need to specify modules like this integration tests? Could we run only tests with integration build flag here and skip the same ones in unit tests?

Reply:
You mean have a way to automatically detect which modules contain a `// +build integration` directive? It's possible with a regex but seems hacky. I don't know of a simple way to do it and didn't want to spend time to optimize too heavily given it's still early days for integrations tests.

Reply:
Don't think we need to worry about skipping anything for unit tests since pretty much all modules should have unit tests.

Reply:
Never mind, you are already doing what I suggested (using --tags). The bash/make code is hard to read for me as must be obvious by now to you :)